brion/types.h

```diff
@@ -50,20 +50,52 @@ class Circuit;
 class CompartmentReport;
 class CompartmentReportPlugin;
 class Mesh;
-class Morphology;
 class SpikeReport;
 class SpikeReportPlugin;
-
 class Synapse;
 class SynapseSummary;
 class Target;
 
+/**
+ * @namespace brion.io
+ * Classes for low level IO.
+ * These classes define the raw interface to access storage as well as the
+ * plugin interfaces to extend the IO capabilities of the library. The plugin
+ * implementations provided by the library are devoted to file based storage.
+ *
+ * @version 1.7
+ */
+namespace io
+{
+class Morphology;
+typedef boost::shared_ptr< Morphology > MorphologyPtr;
+typedef std::vector< MorphologyPtr > Morphologies;
+}
+
+/**
+ * @namespace brion.v2
+ * Classes for future version 2 of the library.
+ * @version unstable
+ */
+namespace v2
+{
+class Morphology;
+}
+
+#ifdef BRION_USE_V2_API
+using namespace v2;
+#else
+using namespace io;
+#endif
+
 typedef vmml::vector< 2, int32_t > Vector2i; //!< A two-component int32 vector
 typedef vmml::vector< 3, float > Vector3f;   //!< A three-component float vector
 typedef vmml::vector< 4, float > Vector4f;   //!< A four-component float vector
```

Comment:
https://github.com/Eyescale/vmmlib/blob/master/vmmlib/vector.hpp#L408 

simply using vmml::Vector2i; should suffice. If some are missing in the above they should be added.